flips/20220817-extensibility.md

```diff
@@ -0,0 +1,423 @@
+`<access modifier> extension <Name> for <Type>: <Interfaces> { ... }`, where the extension methods and fields are declared in the body. As such, 
+the following would be examples of legal declarations of extensions:
+
+```cadence
+pub extension Foo for MyStruct: MyInterface {
+    ...
+}
+
+priv extension Bar for MyResource {
+    ...
+}
+
+priv extension Baz for MyResource: MyOtherInterface, MyThirdInterface {
+    ...
+}
+```
+
+Specifying the kind (struct or resource) of an extension is not necessary, as its kind will necessarily be the same as the type it is extending. At this time,
+extensions can only be defined for a resource or struct composite type. 
+
+The access modifier defines the scope in which the extension can be used: a `pub` extension can extend its original type anywhere that imports it, 
+while an `access(contract)` extension can only be used within the contract that defines it.  Note that this access is different than the access 
+of the fields or methods within the extension itself; a `pub` extension can declare a `priv` field, for example. 
+
+Within the extension, fields and methods can be defined the same way they would be in a struct or resource declaration, with an access modifier and 
+a declaration kind. Any fields and methods on the original type would be accessible to the extension on the `self` value. 
+Note, however, that extensions only have access to the same fields and functions as other code declared in the same place would have; 
+i.e. an extension defined in the same contract as its original type would have access to `pub` and `access(contract)` fields and methods, but not `priv` fields
+or methods, while an extension defined in a different contract and account to its original type would only be able to reference `pub` fields and methods on its 
+original type. Extensions may also not override or overload fields or methods on the original type, even if the original fields are not accessible to the extension. 
+Hence, if the original type `T` declared a `priv` field named `foo`, an extension `E` of `T` would not be able to declare any fields or methods named `foo`, even
+though `foo` is not accessible to `E`. This is to avoid any confusion or ambiguity when reading code. 
+
+Interface conformances are checked against the combination of the original type with the extension applied. In particular, this means that it is possible to declare
+an extension with an interface annotation if neither the extension nor the original type conform to that interface, as long as the resultant extended type would conform. 
+For example:
+
+```cadence
+pub struct interface I {
+    pub fun foo(): Int
+    pub fun bar(): String
+}
+
+pub struct S {
+    pub fun foo(): Int { ... }
+}
+
+pub extension E for S: I {
+    pub fun bar(): String { ... }
+}
+```
+
+In this case, `E`'s declaration conforms to `I` because type of `S` with the extension `E` applied produces a type that conforms to `I`. 
```

Comment:
Was surprised for a moment, then realized `S` does not implement `I`. Maybe add a note to assist the reader